src/arch/host/include/arch/spinlock.h

```diff
@@ -10,8 +10,10 @@
 #ifndef __ARCH_SPINLOCK_H__
 #define __ARCH_SPINLOCK_H__
 
-typedef struct {
-} spinlock_t;
+struct _spinlock_t {
+};
+
+typedef struct _spinlock_t spinlock_t;
```

Comment:
I'd rather either
1. create spinlock_t.h for each arch and just put the original
```
typedef struct {
...
} spinlock_t;
```
there or
2. define `struct spinlock_s` in each spinlock.h and remove typedef from there, and instead put
```
typedef struct spinlock_s spinlock_t;
```
in a separate generic spinlock_t.h and include it wherever needed.

Reply:
That could work, just not sure about the naming `<spinlock_t.h>` itself. I do believe I may have moved a few too many types in this header, hence requested the feedback.